### PR TITLE
Feature/reload ads resize

### PR DESCRIPTION
**Pull request by JGAntunes**
Support for reloading ads on window resize.

awesome gif

### CODE REVIEW

File: packages/ad/gpt.web.js [hunk 1]
```diff
@@ -16,45 +16,48 @@ const getStyles = config =>
...
-      );
-      this.setState({ config }, callback);
-    }
+    const { width } = Dimensions.get("window");
+    this.setState({ width }, callback);
   }
 
   render() {
-    const { config } = this.state;
+    const config = this.config;
```
Comment (trevorah) [inline packages/ad/gpt.web.js]: I know ads are messy, but lets try and ensure the render method only uses this.props and this.state.

File: packages/ad/ad-manager.js [hunk 2]
```diff
@@ -32,15 +37,45 @@ class AdManager {
...
+  }
+
+  // Optional codes argument will unregister only the specified ads
+  unregisterAds(codes) {
+    let slotsToRemove;
+    // We're unregistering all ads
+    if (!codes) {
+      this.initialised = false;
+      this.registeredSlots = {};
+    }
```
Comment (trevorah) [inline packages/ad/ad-manager.js]: Will we ever unregister all ads? Surely a component shouldnt care about the global state...

File: packages/ad/ad-manager.js [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
-  const idx = objIds.indexOf(itemId);
-  return idx > -1 ? obj.filter((item, index) => idx !== index) : obj;
+import gptManager from "./gpt-manager";
+import pbjsManager from "./pbjs-manager";
+
+const defaultOptions = {
+  section: "article",
+  networkId: "25436805",
+  adUnit: "d.thetimes.co.uk"
 };
```
Comment (trevorah) [inline packages/ad/ad-manager.js]: No defaults, they only work for an article component.

File: packages/ad/ad-manager.js [hunk 4]
```diff
@@ -32,15 +37,45 @@ class AdManager {
...
   }
 
-  registerAd(slot) {
-    this.adQueue.push(slot);
+  registerAd(adConfig) {
+    this.adQueue.push(adConfig);
+  }
+
+  // Optional codes argument will unregister only the specified ads
+  unregisterAds(codes) {
```
Comment (trevorah) [inline packages/ad/ad-manager.js]: We only seem to unregister one ad at a time. Why use an array?

**Comment by JGAntunes** [comment]
@trevorah in order to calculate the width of the device window before rendering we're using the `react-native` `Dimensions` API which I assume, in the end, will mean usage of `window` through `react-native-web`. So the answer would be no, I don't think this will work with `renderToString()`... We could do this on the `componentDidMount` but I think that will mean we'll always have a flash between calculating the correct config, get the height and put the placeholder in.

**Comment by trevorah** [comment]
Closing for now so that we can figure out what is the best user experience.

**Pull request closed without merging**